### ISSUE TITLE
arm and disarm causes MATEKH743 to lose connection with radio in INAV 5.1.0

### ISSUE THREAD

**Issue by vlamgat007** [iNavFlight/inav#8409]
## Current Behavior
I am in the process of upgrading a Matek H743-WING V2 Flight Controller + XF Nano Diversity setup from INAV 3.0 to INAV 5.1 and am running into an issue.  When I arm and disarm the system, I lose all radio control to the board (in the Receiver tab the bars stop moving).  I need to reboot the board to gain control.  It does not seem like any of the other switches cause the same behavior.  Firmware for the XF components (RX + VTX) have also been updated.

(FYI....I have successfully completed a similar upgrade for a different plane with a similar setup using Matek F405-WSE Controller + XF Nano and did not experience the same problem)

# Steps to Reproduce
1. Switch on Radio
2. Power up the plane (plug in battery) (the issue exists with and without battery connected)
3. Connect to board with PC via USB
4. Connect to board using INAV 5.1.0
5. Verify that Receiver is connected (all control surfaces moving and bars moving in Receiver tab)
6. Wait for the system to be ready to arm
7. When ready to arm, flip the arm switch
8. Disarm
9. At this point the board stops responding to the radio (bars not moving in the Receiver tab and control surfaces on the plane not moving)
10. Reboot the board to regain control

## Additional context: dump all
<https://pastebin.com/uVBzcK75>

---

# version
# INAV/MATEKH743 5.1.0 Aug 19 2022 / 12:34:02 (76f22b25)
# GCC-10.2.1 20201103 (release)

- FC Board name and vendor:  Matek H743-WING V2 Flight Controller
- INAV version string: 5.1.0

**Comment by b14ckyy** [comment]
what crossfire FW version do you use? There are some older versions with odd behavior like this.

**Comment by b14ckyy** [comment]
hmmm 6.19 should be stable. did you do a factory reset of your crossfire receiver and module after the last update? If you update from 6.14 or earlier you have to do a factory reset. Maybe try that and then see if the issue still appears. 

if yes, I suggest to reset the FC to defautls and just set up the basics and see if this solves the issue. Save a DIFF ALL before that.

**Comment by vlamgat007** [comment]
I did do a factory reset of the TX  but not sure about the RX.  I will test this later today and post my results.
Thanks!

**Comment by b14ckyy** [comment]
Rx you should reset first over the crossfire settings LUA. because you will lose binding and doing it last will be more difficult

**Comment by vlamgat007** [comment]
Thanks, will do.

**Comment by vlamgat007** [comment]
Ok, I think I have narrowed down the issue.  The RX and TX reset did not fix the issue.  As suggested I then reset the defaults on the board and started from scratch. I set up only one mode for arming on Channel 5.
Every time I made a config change and saved it then tested (are + disarm) and verified if I still had radio connection.  I continued till I hit the "**Continuously trim servos on Fixed Wing**" setting.  As soon as I saved this and went through the test cycle, it froze up, and when I removed the setting the issue disappeared.  I tested this multiple times to confirm.

**Comment by b14ckyy** [comment]
this is really strange. can you please post a DIFF of your current config right at this point where enabling or disabling this setting causes the lockup?

**Comment by breadoven** [comment]
Might have something to do with it ?

https://github.com/iNavFlight/inav/blob/69b18f267872b10ceb3327932348a5d0eb6c091d/src/main/fc/config.c#L330

**Comment by breadoven** [comment]
It looks like it suspends the Rx signal for a fixed 1.5s period (`SKIP_RC_ON_SUSPEND_PERIOD` in `rx.c`) before resuming the Rx signal regardless of whether or not eeprom has been saved ... which doesn't make sense. I'm guessing this isn't long enough for 5.1 which takes longer to save eeprom. Perhaps 1.5s is right on the limit for some boards and not others, a timing issue. Although it would make more sense not to have a fixed time period but simply reset the suspend time when eeprom write was finished.

**Comment by breadoven** [comment]
Having said that surely if it's writing to eeprom it shouldn't be able to do anything else should it, like try to process Rx signals ?

**Comment by MrD-RC** [comment]
> 
> @vlamgat007 do you use a switch to arm and disarm or do you disarm with a stick command? 
> 

There is no stick command to disarm anymore. I think it was even gone in 3.0.

**Comment by vlamgat007** [comment]
Diff all at point of failure:  https://pastebin.com/8YEMD5Bg

Please note the Failsafe triggered but I have had multiple times when there was no failsafe but I had no control.

**Comment by vlamgat007** [comment]
> > @vlamgat007 do you use a switch to arm and disarm or do you disarm with a stick command?
> 
> There is no stick command to disarm anymore. I think it was even gone in 3.0.

I use a switch on channel 5 to arm and disarm.

**Comment by vlamgat007** [comment]
diff all with no failsafe:  https://pastebin.com/EDA0WcY8

**Comment by breadoven** [comment]
Failsafe is suspended during eeprom save so it doesn't trigger when the Rx is suspended. The fact that Failsafe behaves erratically during this problem might indicate there is an issue with Rx suspend during eeprom save. Be useful to know why the Rx is suspended during eeprom save, not immediately obvious.

Also wondering if this has anything to do with https://github.com/iNavFlight/inav/issues/7128. The only time I had a Config wipe was on a plane that was connected to Configurator and powered from the battery when it happened. The Rx only works on that plane when on battery power not USB.

**Comment by vlamgat007** [comment]
I have tried with battery connected and without (usb only) and had the same result.

**Comment by breadoven** [comment]
@vlamgat007 Can you try the attached firmware. It's the current master with Rx suspend time increased to 3 seconds. You'll need to use the latest 6.0 Configurator which can be found at http://seyrsnys.myzen.co.uk/inav-configurator-next/.

[inav_6.0.0_MATEKH743.hex.zip](https://github.com/iNavFlight/inav/files/9632264/inav_6.0.0_MATEKH743.hex.zip)

**Comment by 0crap** [comment]
@vlamgat007 If you want to stick with 5.1 you might also want to try my [5.1 build](https://github.com/0crap/inav/releases/tag/5.1.0) without auto ContinuousServoAutotrim saving.

**Comment by vlamgat007** [comment]
Thank you @breadoven and @0crap!  I like having ContinuousServoAutotrim so I will try v6.0.0 first.
I am tied up with work so I will get to this later today.
Thanks!

**Comment by 0crap** [comment]
@vlamgat007 Try whatever you like, but please be assured that in my 5.1 build ContinuousServoAutotrim is fully functional.
Only it will not auto save (which seems to be your issue) the new values after you land and disarm. You can still save the new values if you wish, by going into the OSD and choose save and reboot.

**Comment by vlamgat007** [comment]
@0crap, thanks for the clarification!

**Comment by vlamgat007** [comment]
Success!  I loaded 6.0 and flashed with the 6.0.0 firmware provided by @breadoven.  I updated using the last Diff All that I posted and then tested the arm and disarm.

There is a bit of a longer delay before you get control back after flipping the disarm switch but that seems like a minor issue.

I completed the rest of the setup and everything seems to be working as expected.

BIG THANKS to everyone for your support on this!!

Let me know if you have any further special instructions.  I take it that I need to keep my Matek F405-WSE Controller on 5.1.0?

**Comment by breadoven** [comment]
@vlamgat007  Just remember the 6.0 firmware is dev standard so could have issues, not that I've had any problems. Other FC boards are fine on 5.1.0 if they work OK.

So it appears the default Rx suspend time setting during eeprom write is probably causing problems but it's not clear why it only seems to happen with Auto Trim on disarm but not with other eeprom writes such as sensor detection and Accelerometer calibration on boot up. I assume nothing odd happened previously with 5.1.0 during accelerometer calibration on boot, completed as normal with the beeper confirmation at the end ?

**Comment by b14ckyy** [comment]
@breadoven maybe because when disarming, the actual eeprom write is done first before the actual disarm happens. So all radio inputs are still expected to be valid during that time. All other eeprom saves happen only in already disarmed state.

**Comment by vlamgat007** [comment]
I just came back from flying the plane after the changes.

Bad news is that it froze up when i disarmed after I landed. Good news is that this happened after it was safely on the ground.

**Comment by breadoven** [comment]
> @breadoven maybe because when disarming, the actual eeprom write is done first before the actual disarm happens. So all radio inputs are still expected to be valid during that time. All other eeprom saves happen only in already disarmed state.

It waits until the Arming flag is disarmed before saving. Maybe the problem is Stats also get saved at the same time the Arming flag changes state. 2 eeprom writes in succession. Maybe this needs changing so you only write once on disarm.

**Comment by breadoven** [comment]
Ah, missed the comment that the problem is still there. Probably worth trying @0crap's 5.1 version with the ContinuousServoAutotrim eeprom auto save removed to see if that fixes things.

**Comment by vlamgat007** [comment]
I have loaded the firmware from @0crap.  It has not frozen up yet, unfortunately I ran out of time to do a test flight.

**Comment by vlamgat007** [comment]
Flew and did not hang up on me.  Used the OSD to save.  Strange enough my other wing using the 405-wse controller did hang up after landing and disarming :-|

**Comment by breadoven** [comment]
Would appear this eeprom write on disarm is somehow randomly causing problems.

@vlamgat007  Do you have `stats `set to ON ? Could try turning it OFF to see if that makes a difference. Test on the H743 with current 5.1.0 release that hangs every time you disarm. Stats writes to eeprom on disarm at the same time as the Continuous Autotrim but they're written separately. Hard to see how that could be an issue but maybe the timing causes problems.

Also noticed normal servo Autotrim (not continuous) does a `saveConfigAndNotify()` on disarm rather than just `writeEEPROM()` used for the Continuous Autotrim. Surely the same eeprom save method should be used for both.

**Comment by MrD-RC** [comment]
@breadoven I agree. I think anything that saves on disarm should call saveConfigAndNotify(). Maybe that could just set a flag to do the actual saving and notifying in the next loop, so save is only done once.

**Comment by b14ckyy** [comment]
While things get changes, I guess it might be a good idea to finally also disable autosave for Continous Servo trim. There is no point in saving it every time with deviations of 10us between flights. 
With the longer and longer save times I also see that as a safety issue when the disarm takes 1.5s  or even longer after the disarm switch is applied. This can make the difference between a flesh wound and losing a hand. Just keep it for the classic autotrim and stats as for stats there is no other way and Autotrim is  intentionally used on a single flight.

**Comment by breadoven** [comment]
Saving happens after disarming so it doesn't increase disarm time. You lose the RC link during saving but the motors should have a stopped by then.

**Comment by b14ckyy** [comment]
But the Motors are actually still running. At least on a wing. Not sure about copters as dshot wilkl reset if the signal interrupts briefly during safe.

**Comment by vlamgat007** [comment]
Stats = OFF on both planes

**Comment by breadoven** [comment]
> But the Motors are actually still running. At least on a wing. Not sure about copters as dshot wilkl reset if the signal interrupts briefly during safe.

You're right. I have a delay on disarming for another reason and don't use the servo trim stuff so wouldn't have noticed any additional disarm delays caused by eeprom saves (although I do have Stats enabled).

Seems even though the motors are set to the disarmed state in the mixer code it isn't actually written to the motors until after servoAutoTrim is processed during which the eeprom write will occur:
https://github.com/iNavFlight/inav/blob/c6706adc91a8f6ae0a37163188342240378b2d39/src/main/fc/fc_core.c#L927

This doesn't look like the best way of doing things. Certainly shouldn't have multiple eeprom writes on disarm and the order of events isn't ideal especially if it delays disarming.

**Comment by breadoven** [comment]
> Stats = OFF on both planes

So only 1 eeprom write causes this problem which simplifies things. Only question is why and also why aren't others having the same problem.

Makes you wonder what happens if you turn Stats ON ?

**Comment by vlamgat007** [comment]
> Makes you wonder what happens if you turn Stats ON ?


I will test.

**Comment by MrD-RC** [comment]
@breadoven I've consolidated the save calls in to a single action. Being on the next loop, this should also allow other tasks to complete; such as disarming. Of course, that will depend on the scheduler too.

https://github.com/iNavFlight/inav/pull/8439

**Comment by breadoven** [comment]
> > Makes you wonder what happens if you turn Stats ON ?
> 
> I will test.

Did you get a result from this test in the end ?

**Comment by vlamgat007** [comment]
> Did you get a result from this test in the end ?


I am sorry I have not!  I will see if I can bench test tonight and fly over the weekend.

**Comment by vlamgat007** [comment]
> > > Makes you wonder what happens if you turn Stats ON ?

@breadoven I set stats = ON and it is freezing up again.  This is with the firmware provided by @0crap on the matek h743-wing v2 board.

**Comment by breadoven** [comment]
> > > > Makes you wonder what happens if you turn Stats ON ?
> 
> @breadoven I set stats = ON and it is freezing up again. This is with the firmware provided by @0crap on the matek h743-wing v2 board.

OK, well that clarifies things. Just need to work out what's wrong with the eeprom save now.

**Comment by zvikaf** [comment]
Have the same with H743 Wing V3 ...  #8450

**Comment by 0crap** [comment]
> > > > Makes you wonder what happens if you turn Stats ON ?
> 
> @breadoven I set stats = ON and it is freezing up again. This is with the firmware provided by @0crap on the matek h743-wing v2 board.

Weird stuff.
I bench tested with the `stats = ON` setting on my build and no issues.
That said, it's not using the diff all from vlamgat007, just my own fully configured diff all setup.
After a `arm` and `disarm` the FC stays responsive.
Because autolaunch for fixed wing is always on, the arm command gives the "raise throttle" indication in the OSD.
At that point I disarm and do this over and over, no issues.

**Comment by breadoven** [comment]
> > > > > Makes you wonder what happens if you turn Stats ON ?
> > 
> > 
> > @breadoven I set stats = ON and it is freezing up again. This is with the firmware provided by @0crap on the matek h743-wing v2 board.
> 
> Weird stuff. I bench tested with the `stats = ON` setting on my build and no issues. (MATEKH743 V2) That said, it's not using the diff all from vlamgat007, just my own fully configured diff all setup. After a `arm` and `disarm` the FC stays responsive. Because autolaunch for fixed wing is always on, the arm command gives the "raise throttle" indication in the OSD. At that point I disarm and do this over and over, no issues. (Stats screen shows up on OSD.)

@0crap Are you using the same receiver type, i.e. crossfire ?

**Comment by 0crap** [comment]
> @0crap Are you using the same receiver type, i.e. crossfire ?

I use a ExpressLRS RX (EP1 on v3.0). Which does use the CRSF protocol between RX and FC.

**Comment by MrD-RC** [comment]
> 
> https://github.com/iNavFlight/inav/actions/runs/3169611952
> 
> 
> 
> Is this version OK to test with @MrD-RC ?

Yes, all good.

**Comment by EhAye** [comment]
i'm having a similar problem. H743-WING v2, and tried two different ELRS RX

i can trigger the failsafe by moving yaw to the left or right and holding it there. (0 throttle, pitch&roll middle). This seems to trigger the failsafe a lot more reliably than arm/disarm with a switch... which only causes a failsafe sometimes.

After a failsafe, i tried to reboot the RX and TX (leaving the FC running) by using the elrs lua script. They bind and TX gets full telemetry including GPS telemetry, but failsafe RX flag remains & channels remain unresponsive.

stats=on or off doesn't seem to make any difference, though my vtx isn't powered

**Comment by MrD-RC** [comment]
That sounds like an ELRS issue to me. It is the ELRS receiver that tells INAV it is in failsafe, not the other way around. INAV does not command failsafe, it only does what it is told.

**Comment by breadoven** [comment]
> i can trigger the failsafe by moving yaw to the left or right and holding it there. (0 throttle, pitch&roll middle). This seems to trigger the failsafe a lot more reliably than arm/disarm with a switch... which only causes a failsafe sometimes.

INAV 5.1.0 ? And is this failsafe caused by the yaw stick movement when armed or disarmed or both ?

**Comment by EhAye** [comment]
> INAV 5.1.0 ? And is this failsafe caused by the yaw stick movement when armed or disarmed or both ?

when i arm with a TX switch, the yaw movement does not cause failsafe. It only happens when ch5 switch is disarmed.
yes inav 5.1.0

**Comment by b14ckyy** [comment]
The only stick command that has a centered pitch/roll and a full yaw right input is the Arming Safety bypass but that should not trigger anything else than that. 
isn't one of the yaw stick movements also used for HDZero or Runcam Cameras to enter the camera menu? Is this maybe a ELRS feature that is triggered and reporting a FS to prevent arming while in the menu?

**Comment by breadoven** [comment]
> > INAV 5.1.0 ? And is this failsafe caused by the yaw stick movement when armed or disarmed or both ?
> 
> when i arm with a TX switch, the yaw movement does not cause failsafe. It only happens when ch5 switch is disarmed. yes inav 5.1.0

I'm still not sure what you're saying here. It sounded like a failsafe can be triggered just by moving the yaw stick left or right without touching the arm switch. Are you in fact saying that the failsafe is only triggered when using the arming switch to disarm but the failsafe is more likely to occur with the yaw stick held hi or low when disarming ?

**Comment by EhAye** [comment]
> isn't one of the yaw stick movements also used for HDZero or Runcam Cameras to enter the camera menu? Is this maybe a ELRS feature that is triggered and reporting a FS to prevent arming while in the menu?

It also failsafes on Throttle up, Yaw left, Pitch up Roll Right (both up and to the far corners) I  can't see any commands listed in the documents associated with this position, but it causes a failsafe as well.

The Calibrate Compass stick command causes a failsafe as well (Throttle up, Yaw Right, Pitch down).... most positions don't cause it. But these do.

**Comment by EhAye** [comment]
> I'm still not sure what you're saying here.

sorry. I get failsafes when arming with channel5 switch sometimes.

But while disarmed, if i move the yaw stick to the left and hold it there for about 2 seconds, i get a failsafe. There's a few other stick commands that will cause failsafe as well, like the Calibrate Compass command

**Comment by breadoven** [comment]
> That sounds like an ELRS issue to me. It is the ELRS receiver that tells INAV it is in failsafe, not the other way around. INAV does not command failsafe, it only does what it is told.

If it's related to https://github.com/iNavFlight/inav/commit/3a13edfdad83fb3eaeada195654fbadbe5acb495 then it could be an INAV problem although it's odd it only seems to affect these H743 boards.

**Comment by EhAye** [comment]
> > That sounds like an ELRS issue to me. It is the ELRS receiver that tells INAV it is in failsafe, not the other way around. INAV does not command failsafe, it only does what it is told.
> 
> If it's related to [3a13edf](https://github.com/iNavFlight/inav/commit/3a13edfdad83fb3eaeada195654fbadbe5acb495) then it could be an INAV problem although it's odd it only seems to affect these H743 boards.

While its in the RX failsafe state, i put both the RX and TX into wifi mode and hit the "Save & Reboot" button, to reboot them while keeping the FC powered

They re-establish link and resume advanced telemetry normally. But the FC still reports RX failsafe and shows no channel movements on the flaps or receiver tab 

Could inav just be getting 'stuck' in this state, and refusing to reset the flag?

**Comment by breadoven** [comment]
> > I'm still not sure what you're saying here.
> 
> sorry. I get failsafes when DISARMING* with channel5 switch sometimes.
> 
> But while disarmed, if i move the yaw stick to the left and hold it there for about 2 seconds, i get a failsafe every time. There's a few other stick commands that will cause failsafe as well, like the Calibrate Compass command

I assume it doesn't failsafe when moving the yaw stick if it's never been armed ?

**Comment by EhAye** [comment]
> I assume it doesn't failsafe when moving the yaw stick if it's never been armed ?

freshly rebooting everything, i can simply move the yaw to the left, hold for 2 seconds, and it failsafes. Not touching anything else

**Comment by breadoven** [comment]
> > I assume it doesn't failsafe when moving the yaw stick if it's never been armed ?
> 
> freshly rebooting everything, i can simply move the yaw to the left, hold for 2 seconds, and it failsafes. Not touching anything else

OK, well that changes things if it does it without even disarming.

@vlamgat007 do have the same thing happening just using the stick movements without ever arming ?

**Comment by EhAye** [comment]
I don't know if it means anything, but the "Serial Link Status" indicator flashes Grey a couple times right before the failsafe. Serial Link Status goes back to Blue once the failsafe has activated

**Comment by vlamgat007** [comment]
> @vlamgat007 do you have the same thing happening just using the stick movements without ever arming ?

@breadoven and @EhAye  Not something that I have noticed but it could just be because I do not use that stick command.  I will need to test that specifically.  I will get back to you on this.

Does it make a difference if the fc is usb or battery powered?

**Comment by EhAye** [comment]
> Does it make a difference if the fc is usb or battery powered?

doesn't make a difference for me. Under battery power, no USB it still happens. It is sorta random'ish. Sometimes it failsafes immediately without touching anything.

**BUT it doesn't seem to happen while armed.** (going to test this more tomorrow)


DISARMED: In edgetx telemetry tab i noticed the inav telemetry sometimes halts briefly. This pause happens at the same time the stick/channel inputs become jumpy (feels exactly like a sudden & extreme framerate drop in a videogame). This pause happens every 1 - 5 seconds. Eventually one of these 'pauses' turns into a failsafe and the servos halt.

but the RX's telemetry doesn't miss a beat. It continues uninterrupted even when inav telemetry pauses briefly.

its also not just the 'stick commands' that can cause the failsafe. I was just moving the pitch/roll stick in a circle about half deflection to listen to the servos pausing. It failsafes quite quickly, after maybe 10 or so rotations.



**ARMED:** i'm currently getting no pauses, flight counter just called 11 minutes, no drops in telemetry, no servo pauses, no failsafes, no problems (*so far*)

After 11 minutes i disarmed. Everything seemed "fixed" at first, but then the channel/servo pausing came back after about 15 seconds, and already a failsafe from rotating pitch/roll at 1/3rd deflection after about 25 seconds. 


I've got to leave for the day but i'll be leaving this armed and connected for a couple hours tomorrow to see if i can trigger the failsafe problems while armed

**Comment by EhAye** [comment]
i had left this unplugged and turned off for about an hour.
I was wondering why the problem got progressively worse over time, so i turned it back on and tried it. 
I'm having a hard time making it failsafe again.

could this be a hardware problem? like bad crystal oscillator issue on the matek RX? could it be a bad oscillator that's heating up and changing frequency?

its even surviving the yaw-left stick command that kills it 90% of the time. Just a minor hickup in servo movement when i do it

**Comment by EhAye** [comment]
I've had this armed on Battery (no USB) for over 2 hours and no problems with anything at all. No hickups on the flaps, no telemetry issues, no problems.

I tried simulating a failsafe by turning off my radio. the FC went into failsafe mode and spun up the motor. When link re-established, i was able to regain control and everything is still working fine.

I'm worried i'll get stuck in this "bugged failsafe" during a real flight.. But i can't seem to trigger the failsafe while its armed on battery. I guess that might rule out hardware problems?

**Comment by jayss66** [comment]
I am having this exact same problem. Crossfire receiver, Matex H743 wing V1, running INAV 6.0. I was just about to go to the flying field, I thought I had this problem solved last night as it didn't do it then when the battery was plugged in. I was double checking just now before I loaded it into the truck, and it did it again on disarm with only the battery plugged in. I will disable continuous trim servos for now.
Thanks.

**Comment by MartinHugh** [comment]
I experienced a failsafe with a H743 + iNav5.1 yesterday as I disarmed.

**Comment by b14ckyy** [comment]
@MartinHugh do you by chance have Autolevel on together with ANGLE mode? Maybe post a DIFF please

**Comment by MartinHugh** [comment]
> @MartinHugh do you by chance have Autolevel on together with ANGLE mode? Maybe post a DIFF please

Diff at : https://pastebin.com/Nn5Sx2CL
AutoLevel and ANGLE *are* ganged together.

**Comment by b14ckyy** [comment]
I think Autolevel also autosaves. And also this is a tuning mode and should not be used constantly. 
maybe one of the devs can check if it autosaves, i am not 100% sure.

**Comment by MartinHugh** [comment]
> I think Autolevel also autosaves. And also this is a tuning mode and should not be used constantly. maybe one of the devs can check if it autosaves, i am not 100% sure.

In case this is relevant : I am not convinced I switched into Angle/autolevel on both flights (only one).  And when I did, it was only for 10s early on in the flight.  Prior to disarming I was in Manual mode or Acro in both cases.  I did not trigger a save from the sticks or OSD either at any time.

**Comment by breadoven** [comment]
Doesn't auto save from what I can see and also the Wiki states it needs to be manually saved.

But based on what @EhAye found this doesn't seem to be related to disarming anyway. Instead it appears to be related to just being in the disarmed state. It could happen immediately on disarming or perhaps by just by moving the sticks to the extremes when disarmed. Eeprom saves perhaps just increase the likelihood of it happening on disarm.

**Comment by vitaly-rudenya** [comment]
Same problem for me.
For my case I've found that it's stops working in case saving !! flight modes !!
After that reciever tab shows static values.
Rebooting only FC (reciever keeps powered) solve the problem in a time
Sime time it happend when just moving stics on my radio
H743 V3 + ELRS-R24-D

**Comment by breadoven** [comment]
> For my case I've found that it's stops working in case saving !! flight modes !!

You mean changing and saving flight mode settings in Configurator causes the FC to lose the receiver signal ?. Does this also cause a failsafe ?

What causes the failsafe on INAV 5.0 exactly, it's always in Failsafe or only after doing certain things ?


I had a look at this trying to see why it only appears to happen when disarmed and couldn't find anything obvious. There doesn't seem to be anything that changes when disarmed that would affect the receiver signal or CRSF processing. So a bit baffling really as to why this is happening.

Would be useful to know if it also affects receiver protocols other than Crossfire.

**Comment by MartinHugh** [comment]
OK, so I am not sure exactly what parameters  we should be looking at - we may be describing different fault causes in this thread, so all I can do is report what I have seen.

As I mentioned above, I have reported multiple cases of failsafe immediately following disarm.  This is not reliably repeatable. But it happened twice in the same flying session. 

However following on from the recent posts by @vitaly-rudenya and @breadoven above.  I happened to be reconfiguring my modes setting in configurator tonight.  No battery fitted, FC just connected to USB, and transmitter on to test the positions of the mode selections on the inav screen.

If, after making some small changes to the sliders I press "Save" on that tab in the configurator and then immediately move switches on the transmitter (e.g. in my case I was setting flight modes on SA/SB), the  the modes animation freezes (sliders do not move) Also the receiver tab in configurator in static/frozen too.  Nothing responds to the transmitter switches or gimbals.  This seems to be repeatable.

Also of interest.  
- The GPS tab in configurator is still showing updates, which probably indicates that the FC MCU is still scheduling normally
- Telemetry is still being sent back from the RX to the TX (i.e. values are changing)
- The RX LED is indicating that it is bound to the TX

I am running
MATEKH743
iNav 5.1.0
ELRS 3.0

**Comment by EhAye** [comment]
@MartinHugh I had similar issues when configuring my modes, saving, and switching to the receiver tab to check stick inputs.
Sometimes it would failsafe while flipping between the tabs. Im not sure if it was the tab switch, or saving the mode page that triggered it... sometimes it happens shortly after switching to the reciever tab...

It also doesn't happen 100% of the time. Sometimes i can save and switch tabs just fine. But it will quickly failsafe within a minute or two, and its USUALLY triggered by me doing something (switching tabs, moving sticks, saving settings etc). Although occasionally it will failsafe automatically after bootup.

@vitaly-rudenya same with me. If i reboot the RX, the failsafe persists. But if i reboot the FC while keeping the RX on, the failsafe is cleared. The whole time i am receiving telemetry from both ELRS and iNAV.

Feels like the FC is getting bugged in a fake failsafe state... but only for this one processor?

**Comment by MartinHugh** [comment]
> @MartinHugh I had similar issues when configuring my modes, saving, and switching to the receiver tab to check stick inputs. Sometimes it would failsafe while flipping between the tabs. Im not sure if it was the tab switch, or saving the mode page that triggered it... sometimes it happens shortly after switching to the reciever tab...

Interesting, In my case it I was not switching tabs until after the freeze had occurred, so for me at least the tab switching was not the trigger.

**Comment by vitaly-rudenya** [comment]
> e

So there is no fail save due-to values update stop.

But permantly catcing serial loss and FS on INAV configurator when ELRS is connecting.
With Arducopter with same hardware all works fine

**Comment by MartinHugh** [comment]
> > * Telemetry is still being sent back from the RX to the TX (i.e. values are changing)
> 
> Does this include FC related telemetry or just the native Rx telemetry (signal info LQ, RSSI etc) ?
> 
> From what @vitaly-rudenya reports it does seem to be CRSF related. Only reason CRSF would cause a failsafe is if there are no packets being received by the FC or the packets are bad (CRSF doesn't use a failsafe flag from what I can glean, just uses no pulses on failsafe).

Great question. I have just checked.
The fault induced first time.  Very reproducible.

At the point that the configurator GUI first freezes, all the telemetry displayed in the TX freezes and is displayed red - I would infer this as a failsafe condition, though I have not warning or other indication of this from the configurator.

Then immediately after the telemetry all springs back into life. RX telemetry + FC telemetry.
But the configurator remains frozen and moving TX switches has no effect on it

**Comment by MartinHugh** [comment]
> From what @vitaly-rudenya reports it does seem to be CRSF related. Only reason CRSF would cause a failsafe is if there are no packets being received by the FC or the packets are bad (CRSF doesn't use a failsafe flag from what I can glean, just uses no pulses on failsafe).

@breadoven, is your hypothesis then that the saving of the configuration causes a delay in the RX UART being serviced, such that the FC decides it has failsafed ?

That is one thing, but the TX-RX link does reestablish, so why does the configurator not unfreeze ?

I am clutching at straws here :-)

**Comment by breadoven** [comment]
What happens if Telemetery is disabled ?
Also does setting failsafe_delay to the maximum (20s) make any difference ?

**Comment by MartinHugh** [comment]
> What happens if Telemetery is disabled ? Also does setting failsafe_delay to the maximum (20s) make any difference ?

Telemetry disabled, makes no difference.  Possible to cause lockup in configurator almost instantly

failsafe delay changed from 5ds to 200ds.  
The first immediate freeze seemed to recover, but the one which followed it a second or so later did not.

**Comment by EhAye** [comment]
> The first immediate freeze seemed to recover, but the one which followed it a second or so later did not.

I think this is the same freeze i get, when i'm rotating my pitch/roll stick at half deflection around and around smoothly to listen/watch the flaps, and i'll get a momentary pause/hickup in the servo response, which recovers after a second. If i watch on the receiver tab, the channel animations pause along with the flaps.

It will do this 1, 2, maybe 3 times. Sometimes it recovers, but a failsafe will eventually happen during one of these pauses.

I don't think its my stick movement that's causing it. I think there's just a periodic hiccup in the flight controller, and sometimes (usually) it causes a failsafe. I'm not sure what starts the hickups. It seems like doing anything can start them up.

Once that failsafe happens, the ELRS + FC telemetry continue to transmit for a while. I've noticed the FC telemetry drops out after 5 or 10 or 20 minutes... there's no set time... ELRS telemetry never stops.

**Comment by EhAye** [comment]
**This is really weird.**

In the ELRS LUA script i DISABLED telemetry.
In INAV Configuration tab, i DISABLED Telemetry.

I'm able to rotate my pitch/roll stick and everything seems fine.
I move Yaw Left and hold it there for 2 seconds, no problem. (a few days ago, Yaw Left would reliably trigger a failsafe)
I move Yaw Right and hold it there for 2 seconds, no problem.

I leave ELRS LUA telemetry DISABLED 
I ENABLE inav telemetry in the configuration tab:

I'm able to rotate pitch/roll stick and everything seems fine.
I move Yaw Left and hold it there for 2 seconds, no problem.
I move Yaw Right and hold it there for 2 seconds, Failsafe. Repeatable.
(throttle position doesn't matter. I tried Throttle Up, Middle, Down, 1/4th. As long as Yaw-Right is held for a second, it failsafes)

**Turning on and off inav telemetry, i am able to turn this Yaw-Right failsafe trigger on and off.** I powered off all my hardware, let it rest for 5 minutes, then tried again. Same result. I can reliably turn the Yaw-Right failsafe off by turning inav telemetry off. Turning inav telemetry back on will enable the Yaw-Right failsafe.


**EVEN STRANGER: Currently my Yaw is resting at 1498 on the receiver tab. If i very carefully nudge Yaw right to 1501 it will failsafe. I believe some other failsafes I've experienced is due to stick jitter nudging it over.**

I changed EdgeTX's PPM Center for the Yaw channel to 1505. iNav's Reciever tab shows it at 1503 and it failsafed.

I kept EdgeTX PPM Center at 1505, and rebooted the Flight Controller. It appears to be stable., even though Yaw is resting at 1503 currently. **I can now move Yaw around on the Right side no problem!**

I move Yaw Right and hold it there for 2 seconds, no problem!
I move Yaw Left and hold it there for 2 seconds, FAILSAFE! (Yaw at 1478 on receiver tab)


To to be clear:

- If my FC boots up with Yaw 1505, i can move Yaw around to the right without problem, but moving to the left will failsafe.
- If my FC boots up with Yaw at 1498, i can move Yaw around on the Left without problem, but moving it to the right will failsafe.

Disabling iNAV telemetry seems to solve MOST of this problem. 
With iNAV telemetry DISABLED, i am able to move each individual stick around to its extremes, holding it wherever i like, no problem.

HOWEVER, some other stick-combo positions will still cause problems, Even with inav telemetry Disabled OR enabled:
Throttle down, Yaw left, Pitch down Roll right = the usual failsafe

Throttle Up, Pitch Down = **MAJOR LOCKUP.** This stick position failed differently. The failsafe parachute does not turn red, instead "Serial Link Status" repeatedly blinks, indefinitely. Different configurator tabs load extremely slowly if at all. Disconnected the configurator and tried to reconnect, it can't reconnect
First reboot did not seem to fix it. Second reboot i was able to connect but the configuration was wiped out.

Upon restoring my [diff](https://pastebin.com/L9LwZxit), i am able to reproduce all of the above problems in this post... 

I was also able to reproduce the major lockup... this time it took a few extra reboots before it came back to life.

I've just re-flashed (5.1 still) with full chip erase option. Restored diff. All of the above problems in this post are still repeatable.

**Comment by breadoven** [comment]
@EhAye is this still only happening when Disarmed but never when Armed ?

**Comment by EhAye** [comment]
@breadoven yep, i've just armed it and tried. There is no hickups or failsafes when it is armed. 
upon disarm there was no failsafe.
after i disarmed, i moved yaw left, and it failsafed. (my yaw is still around 1503 resting/at bootup)

While disarmed, powered via USB, the yaw-left thing causes failsafe about 95-99% of the time.
While disarmed powered via BATTERY only (no USB), the yaw-left thing causes a pause/freezing of the channels (visible on the flaps) about 75% of the time. and results in a failsafe about 25% of the time. If i continue holding the yaw to the left, it will failsafe within 5 seconds every time... as if the pauses accumulate or become too long

While armed, yaw doesn't cause any problems, no failsafes, no pausing/freezing.

i wonder if people's upon-disarm failsafes are caused by the 1500 yaw thing? if they hold their stick to the left or right when they disarm, i'd bet only one direction will cause failsafe upon disarm

**Comment by breadoven** [comment]
@EhAye can you provide a DIFF of your settings, not sure you did it before above.

**Comment by EhAye** [comment]
> can you provide a DIFF of your settings

https://pastebin.com/t1HZKSMs is what i'm currently using

I also edited my previous post to add some information about the "tasks" command and the "TELEMETRY" row.

**Comment by breadoven** [comment]
@EhAye Looking at the stick movements you mention some of them are stick commands so intended to do something although they shouldn't cause lockups:
https://github.com/iNavFlight/inav/blob/master/docs/Controls.md.

The Telemetry task info suggests the Telemetry is hanging for some reason and blocking the receiver signal possibly ?. 36,555 microseconds is way too long to process the task. Needs someone who better understands how CRSF works to really understand what is going on. You'd think the fact it only happens when disarmed would help narrow down the cause but it's not obvious what difference that makes.

**Comment by EhAye** [comment]
@breadoven whats even stranger is it seems to only effect some boards? Is EVERYONE with a "h743 wing v2" having this issue? or only some of us?
I know lots of people are claiming to use H743 successfully... maybe they're using the -slim versions? I haven't seen anyone complain about the h743-slim yet, only the -wing variants.

Because of the hardware difference in success rate, it makes me wonder if there's not some deeper hardware bugs. But then people don't seem to have a problem with ArudPilot on the h743-wing-v2

What differences in software/hardware is there between the -wing and -slim? maybe there's just some bad crystal oscillators or pullup resistors on the -wing variants?

I wish i had continued programming when i was younger. I wasn't too bad at it. Although now 20 years later i'd be hard pressed to write a lua script

**Comment by MartinHugh** [comment]
> Because of the hardware difference in success rate, it makes me wonder if there's not some deeper hardware bugs. But then people don't seem to have a problem with ArudPilot on the h743-wing-v2
> 
@breadoven, just for the record, my problems are with the H743-wing-V3 not V2

**Comment by MrD-RC** [comment]
The H743-SLIM and H743-WING use the same target. In fact, all the Matek H743 flight controllers use the same firmware target. MATEKH743

**Comment by EhAye** [comment]
I tried unplugging the USB/beeper daughterboard and it made no difference. 

I did notice the blue LED stops blinking while the channel/flap freezing is happening. The blue LED will stay off, or stay on (whichever it happened to be) for the duration of the freeze, and resume blinking after the freeze. After failsafe happens, it continues blinking normally.
(normally there is a solid red LED and a blinking blue LED while disarmed)

Do we know of anyone who has a wing-v2 or wing-v3 without these problems?

**Comment by EhAye** [comment]
### R:I:I:P
Progress (ish!)

I noticed this post over here https://github.com/ExpressLRS/ExpressLRS/issues/1818 It sounded awfully familiar. He doesn't mention his Flight Controller name. But he did mention the problem goes away after disabling ELRS VTX Administrator. 

I tried disabling the VTX Administrator and poof. No more failsafes. I can move my yaw stick past 1500 while disarmed. No hickups, no pausing, no failsafes.
I re-enable the VTX Administrator (R:I:I:P) and i get the yaw failsafes again.

So i currently have VTX Administrator Disabled, and i can re-enable inav telemetry, and all the problems appear to have vanished.

**Comment by MartinHugh** [comment]
> 

Fantastic find @EhAye.

Which subsystem does this point to?
Is this actually an ELRS problem ?

**Comment by breadoven** [comment]
Well that would explain why this problem only happens when disarmed given VTX Administrator will only send data when disarmed:

https://www.expresslrs.org/2.0/quick-start/transmitters/lua-howto/#vtx-administrator

Not sure how this works though in relation to the flight controller controlling VTX settings.

**Comment by MartinHugh** [comment]
> ### R:I:I:P
> Progress (ish!)
> 
> I noticed this post over here [ExpressLRS/ExpressLRS#1818](https://github.com/ExpressLRS/ExpressLRS/issues/1818) It sounded awfully familiar. He doesn't mention his Flight Controller name. But he did mention the problem goes away after disabling ELRS VTX Administrator.
> 
> I tried disabling the VTX Administrator and poof. No more failsafes. I can move my yaw stick past 1500 while disarmed. No hickups, no pausing, no failsafes. I re-enable the VTX Administrator (R:I:I:P) and i get the yaw failsafes again.
> 
> So i currently have VTX Administrator Disabled, and i can re-enable inav telemetry, and all the problems appear to have vanished.

Can I check *how* you are actually disabling vtx assistant.  There does not seem to be an obvious was in the LUA.  Is this a compilation setting ?

**Comment by EhAye** [comment]
Well it didn't solve all of the failsafes. 

Throttle Down, Yaw Left, Pitch Down, Roll Right (Both down, and apart) will still consistently trigger the failsafe. (even with inav telemetry disabled, elrs vtx administrator disabled)

Maybe this is a different/separate failsafe issue? Because this IS a stick command that triggers "Save Setting". And this failsafe doesn't happen until both sticks are all the way to the corners and held there.

**Comment by MrD-RC** [comment]
That is a separate issue that needs to be looked in to. There needs to be a way to stop stick commands when connected to Configurator. There is already an issue created with that problem. 

The problem in _this_ issue is down to ELRS. If you turn off their feature, the problem stops. This is why we don't see this problem with any other RC links. It is strange how it seems to only affect H743 though.

**Comment by vlamgat007** [comment]
> this problem with any other RC links

@MrD-RC , @breadoven, @EhAye ,

I might be missing the plot seeing as this thread has already evolved quite a bit, but just wanted to point out that my original issue is with using a crossfire nano diversity receiver.

**Comment by MrD-RC** [comment]
@vlamgat007 it does seem like 2 issues have been mixed here. The original issue may be fixed with #8439. Did you manage to try the firmware @breadoven linked to [here](https://github.com/iNavFlight/inav/issues/8409#issuecomment-1271248069)?

@EhAye added a different problem to this issue. Which does seem to be caused by an ELRS "feature". Disabling the feature stopped the failsafe after disarm.

I think the good thing to do would be focus on the lockup on saving after a disarm. Potentially because of the EEPROM saves.

**Comment by vlamgat007** [comment]
Thank you @MrD-RC, somehow I missed that.  I loaded firmware provided by @0crap also in an earlier comment on this issue which is what I am using currently and has been working well (https://github.com/0crap/inav/releases/tag/5.1.0)

I will definitely test the firmware suggested by @breadoven.  I apologize for dropping the ball on this!

**Comment by MrD-RC** [comment]
No worries @vlamgat007. It's good to get it back on track after the ELRS detour 🤣

It will be interesting to hear how that firmware behaves.

**Comment by zvikaf** [comment]
Just to clear things and amplify my understanding : 
(1) writing to the flash memory made TBS ( /LRS ?)  communication hang, did this phenomena accrue with other radio links ? 
(2) seems to happened only at the 743 family of controllers, did it happened on other families ?   
TIA

**Comment by breadoven** [comment]
> > > > > Makes you wonder what happens if you turn Stats ON ?
> > 
> > 
> > @breadoven I set stats = ON and it is freezing up again. This is with the firmware provided by @0crap on the matek h743-wing v2 board.
> 
> Weird stuff. I bench tested with the `stats = ON` setting on my build and no issues. (MATEKH743 V2) That said, it's not using the diff all from vlamgat007, just my own fully configured diff all setup. After a `arm` and `disarm` the FC stays responsive. Because autolaunch for fixed wing is always on, the arm command gives the "raise throttle" indication in the OSD. At that point I disarm and do this over and over, no issues. (Stats screen shows up on OSD.)

@0crap Looking back through this stuff I noticed your post which indicated your H743 FC wasn't affected by the eeprom write problem. However, checking the code it seems there is a 10 second minimum arming time before STATS is saved on disarm. Do you think you exceeded this 10s limit when you tested it ?

One possibility here to help debugging is a custom firmware with debugging added to the Rx/Failsafe code to try and work out what the signal is doing when it locks up. However, given this happens when disarmed there won't be any log with the debug info so the debugging would need to be taken from the Configurator sensors tab or recorded from the OSD (probably not DJI unless the WTFOS hack is used ?).

**Comment by breadoven** [comment]
The attached firmware includes debugging on the Rx and Failsafe side as follows:

0 - are flight channels valid (10 = invalid, 11 = valid)
1 - is Rx signal received (10 = no, 11 = yes)
2 - is Failsafe receiving Rx data (10 = no, 11 = yes)
3 - time in us entering Rx/Failsafe checking function
4 - time delta between Rx update checks (rxUpdateCheck)
5 - position indication in rxUpdateCheck related to Rx frame status
6 - checks if suspendRxSignal (value 10) or resumeRxSignal (value 11) functions run during eeprom save

`set debug_mode = ALWAYS` and debug values should be shown in the Configurator Sensor tab or available on the OSD if you know how to set it up.
[inav_6.0.0_MATEKH743.hex.zip](https://github.com/iNavFlight/inav/files/9928162/inav_6.0.0_MATEKH743.hex.zip)

**Comment by vlamgat007** [comment]
Thank you @breadoven I will load and test this evening.

**Comment by 0crap** [comment]
> @0crap Looking back through this stuff I noticed your post which indicated your H743 FC wasn't affected by the eeprom write problem. However, checking the code it seems there is a 10 second minimum arming time before STATS is saved on disarm. Do you think you exceeded this 10s limit when you tested it ?
> 

Probably not, 10 sec is a long time if you have to wait for it. :-)
So I did a retest with my own [build](https://github.com/0crap/inav/releases/tag/5.1.0).
STATS is set to ON and armed for more then 10 seconds on the bench.
After disarm all stays responsive, no issues. Repeated multiple times.
(If it matters, the slider to always enable autolaunch is ON, but armed is armed, motor was spinning at idle rpm.)

**Comment by vlamgat007** [comment]
> @vlamgat007 it does seem like 2 issues have been mixed here. The original issue may be fixed with #8439. Did you manage to try the firmware @breadoven linked to [here](https://github.com/iNavFlight/inav/issues/8409#issuecomment-1271248069)?

@MrD-RC and @breadoven I tested the firmware quoted above and I consistently trigger failsafe.  Battery and usb irrespective.  Exact same behavior as originally reported.

**Comment by vlamgat007** [comment]
@breadoven  when I load the debug firmware version with 6.0.0 fp2  I loose connection to the sensors.  See the screenshot.
![No Sensors](https://user-images.githubusercontent.com/114024021/200138134-4b55ed06-4055-41e1-8ab9-706d55d2d209.JPG)

**Comment by vlamgat007** [comment]
@breadoven and @MrD-RC Seems for now I will have to revert to the 5.1 firmware from @0crap from Sep 23 comment: 

If you want to stick with 5.1 you might also want to try my [5.1 build](https://github.com/0crap/inav/releases/tag/5.1.0) without auto ContinuousServoAutotrim saving.

**Comment by b14ckyy** [comment]
@vlamgat007 
that's expected. There where changes in INAV in the latest master that also need a Configurator update. Get the last nightly configurator from here and all will be okay again http://seyrsnys.myzen.co.uk/inav-configurator-next/

**Comment by Ktr128** [comment]
I have the same problem.
Matek H743-WING V3, INAV 5.1.
The elevon servos hung during autolaunch and the flying wing crashed. Although before that it took off 3 times perfectly.
After that, I caught the elevons hovering while testing in my hand.
Turned on the ANGLE mode and moved the flying wing in pitch.
After about 30 seconds, the elevons hung for 1-2 seconds, after which they started working again.
Has anyone dealt with this problem?

**Comment by breadoven** [comment]
Nobody have any luck using the "debugging" firmware listed above ?

**Comment by vlamgat007** [comment]
I loaded it with 6.0 and lost connection with all my sensors.  I then reverted and did not try again. I have been told to use the latest "nightly version" of configurator which should fix the issue  it I have not got around to it.  I will see if I can do it today.

**Comment by vlamgat007** [comment]
> set debug_mode = ALWAYS



> Nobody have any luck using the "debugging" firmware listed above ?

Update: I removed the crossfire TX, RX, and VTX and switched my system to ELRS.  I am now running a RMRC 1.3ghz vtx.  Just mentioning these changed parameters.

I **could not** trigger a failsafe.  Attached a the debug sensor logs.
 
![Debug 1](https://user-images.githubusercontent.com/114024021/201448632-4f05f2da-307a-4a23-8fff-2ac99c619483.jpg)
![Debug 2](https://user-images.githubusercontent.com/114024021/201448644-775fc1c1-d649-46ae-a123-7bb94404086d.jpg)
![Debug 3](https://user-images.githubusercontent.com/114024021/201448647-6877f4ae-a4ad-4ec7-a876-4e3d2f55992b.jpg)

**Comment by breadoven** [comment]
Thanks for that @vlamgat007 although not so useful since there was no failsafe so everything behaved as expected. And this would appear to be because you've switched RC gear albeit to ELRS which also uses CRSF ... strange it now seems to be behaving.

I assume there was a disarm or eeprom save where debug 6 changed value @ around 135 s ?

Also what settings did you change switching to ELRS ?

**Comment by vitaly-rudenya** [comment]
From my side I've tried on Clean setup (after chip clean reflash).
UART 6 was configures as RX
On Radio tab: Serial CRSF
No other configurations were made and no other devices connected. Only board itslef was connected to PC through USB

**Comment by OptimusTi** [comment]
Just tested MATEKH743 INAV 5.1 and 6.0 master. No issues with TBS CRSF 6.19.

**Comment by mrbigglesw0rth** [comment]
I have a similar issue, but with a different board: OMNIBUSF4V3
Upon disarm, I hear the ESCs startup beep and see the servos not reacting for ~1.5 sec. 
In the configuratior, I see the "MSP round trip" goes up to triple digits.
After this short period, everything seems to be fine. No power cycle needed. Just the ESC reboot give me a very bad vibe.

Workaround: disable continuous servo trim OR change ESC protocol to anything but DSHOT.

As there were no 6.0 firmwares posted in this thread for my board, I could not see if this will be fixed. Is there a way to get nightly firmwares for any board?

**Comment by b14ckyy** [comment]
@mrbigglesw0rth this is normal during safe procedure. 
it will be optimized in the next 6.0 released by https://github.com/iNavFlight/inav/pull/8439

has nothing to do with the bug here.

**Comment by vlamgat007** [comment]
Just an update after installing INAV 6 Horizon Hawk.  Same setup as original post but now running ELRS.  The system consistently looses RC connection on disarm.  If I switch off "Continuously trim servos on Fixed Wing" the connection is not lost and the problem disappears.

**Comment by breadoven** [comment]
> Just an update after installing INAV 6 Horizon Hawk. Same setup as original post but now running ELRS. The system consistently looses RC connection on disarm. If I switch off "Continuously trim servos on Fixed Wing" the connection is not lost and the problem disappears.

Would be useful if you could try https://github.com/iNavFlight/inav/pull/8907.

(should be possible to use the firmware shown in the Artifacts section of https://github.com/iNavFlight/inav/actions/runs/4500418426 if you can't compile yourself).

**Comment by vlamgat007** [comment]
> > Just an update after installing INAV 6 Horizon Hawk. Same setup as original post but now running ELRS. The system consistently looses RC connection on disarm. If I switch off "Continuously trim servos on Fixed Wing" the connection is not lost and the problem disappears.
> 
> Would be useful if you could try #8907.
> 
> (should be possible to use the firmware shown in the Artifacts section of https://github.com/iNavFlight/inav/actions/runs/4500418426 if you can't compile yourself).

@breadoven  thanks.  I will test tonight (CST).  Can you confirm, I downloaded inav-6.1.0-ci-20230323-76d2769.zip but there was also a zip file in a post today by @DzikuVx [inav_6.0.0_MATEKH743.zip](https://github.com/iNavFlight/inav/files/11101435/inav_6.0.0_MATEKH743.zip)

Which one should I use?  Thanks!

![image](https://user-images.githubusercontent.com/114024021/228583446-4e423b41-7839-48e5-9ead-7e92ef4ea553.png)

**Comment by breadoven** [comment]
Either should work but @DzikuVx hex would be best given it should definitely be the right one.

**Comment by vlamgat007** [comment]
t

> Either should work but @DzikuVx hex would be best given it should definitely be the right one.

Thank you @breadoven and @DzikuVx

**Comment by vlamgat007** [comment]
> t
> 
> > Either should work but @DzikuVx hex would be best given it should definitely be the right one.
> 
> Thank you @breadoven and @DzikuVx

@breadoven and @DzikuVx I tested both the following files

- inav_6.0.0_MATEKH743.hex
- inav_6.1.0_MATEKH743_ci-20230323-76d2769.hex

The 6.0.0 file did not work, I was able to initiate failsafe consistently.  Not every time but probably >50% of the time.
The 6.1.0 file seems to work perfectly, I was not able to initiate failsafe at all.

**Comment by 0crap** [comment]
@vlamgat007 can you try the hex provided here please?

_Originally posted by @DzikuVx in https://github.com/iNavFlight/inav/issues/8905#issuecomment-1495645483_

**Comment by dcan999** [comment]
I am getting this issue on 7.0.0. It doesn't lose RX link every time you arm/disarm with continuous trim enabled, it lets me arm/disarm 4 or 5 times before it loses RX link. With continuous trim off, it never loses link.
